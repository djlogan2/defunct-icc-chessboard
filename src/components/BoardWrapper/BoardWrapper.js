import React from 'react'

import styles from './boardWrapper.module.css'

const BoardWrapper = ({ size, children }) => (
  <div style={{ width: size, height: size }} className={styles.boardWrapper}>
    {children}
  </div>
);

export default BoardWrapper
