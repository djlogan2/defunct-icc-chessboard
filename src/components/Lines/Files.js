import React from 'react'
import {
  FILES_ARRAY,
  FILES_BOTTOM_SIDE,
  FILES_DISABLED,
  FILES_TOP_SIDE
} from '../../constants/boardConstants'
import { WHITE_PLAYER_PERSPECTIVE } from '../../constants/systemConstants'

const Files = ({ width, side, perspective }) => {
  if (side === FILES_DISABLED) {
    return null
  }

  return (
    <div
      style={{
        width,
        bottom: side === FILES_BOTTOM_SIDE && 0,
        top: side === FILES_TOP_SIDE && 0,
        position: 'absolute',
        display: 'flex',
        flexDirection:
          perspective === WHITE_PLAYER_PERSPECTIVE ? 'row' : 'row-reverse',
        justifyContent: 'space-around'
      }}
    >
      {FILES_ARRAY.map((file, index) => (
        <p key={index}>{file}</p>
      ))}
    </div>
  )
}

export default Files
