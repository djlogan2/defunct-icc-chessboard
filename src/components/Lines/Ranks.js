import React from 'react'
import {
  RANKS_DISABLED,
  RANKS_LEFT_SIDE,
  RANKS_RIGHT_SIDE
} from '../../constants/boardConstants'
import { WHITE_PLAYER_PERSPECTIVE } from '../../constants/systemConstants'

const Ranks = ({ width, height, side, perspective, ranks }) => {
  if (side === RANKS_DISABLED) {
    return null
  }

  return (
    <div
      style={{
        width,
        height,
        left: side === RANKS_LEFT_SIDE && 0,
        right: side === RANKS_RIGHT_SIDE && 0,
        position: 'absolute',
        display: 'flex',
        flexDirection:
          WHITE_PLAYER_PERSPECTIVE === perspective
            ? 'column-reverse'
            : 'column',
        justifyContent: 'space-around'
      }}
    >
      {ranks.map((rank, index) => {
        return (
          <p
            style={{ textAlign: side === RANKS_LEFT_SIDE ? 'right' : 'left' }}
            key={index}
          >
            {rank}
          </p>
        )
      })}
    </div>
  )
}

export default Ranks
