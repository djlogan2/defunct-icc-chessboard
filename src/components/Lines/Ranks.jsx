import React from 'react'
import {
  RANKS_ARRAY,
  RANKS_DISABLED,
  RANKS_LEFT_SIDE,
  RANKS_RIGHT_SIDE
} from '../../constants/boardConstants'
import { WHITE_PLAYER_PERSPECTIVE } from '../../constants/systemConstants'

const Ranks = ({ height, side, perspective }) => {
  if (side === RANKS_DISABLED) {
    return null
  }

  return (
    <div
      style={{
        height,
        left: side === RANKS_LEFT_SIDE && 0,
        right: side === RANKS_RIGHT_SIDE && 0,
        position: 'absolute',
        display: 'flex',
        flexDirection:
          WHITE_PLAYER_PERSPECTIVE === perspective
            ? 'column-reverse'
            : 'column',
        justifyContent: 'space-around'
      }}
    >
      {RANKS_ARRAY.map((rank, index) => {
        return <p key={index}>{rank}</p>
      })}
    </div>
  )
}

export default Ranks
