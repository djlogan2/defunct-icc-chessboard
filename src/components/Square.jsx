import React from 'react'
import PropTypes from 'prop-types'
import LegalMoveDot from './LegalMoveDot'
import SquareOrCircle from './SquareOrCircle'
import RankAndFileIndicator from './RankAndFileIndicator'
import Piece from './Piece'

// OK, firstly, no images, no pieces, no colors, no styles, no CSS
// in this library. Since we are supposed to be building this to allow
// total customization, my idea is that at the very top level, we pass in
// an object that contains EVERYTHING EVERY component needs to handle itself.
// These classes will do things like shrink, enlarge, crop, center, whatever it
// needs to do, but it will NOT decide what things look like. That is all passed in.
//
// I'm thinking like this:
// <Board styles={styles_object} />
//   Then every component, all the way down, gets passed the same things. See below.
//
// Nextly, you can see how I am thinking about putting this together. This square, for example,
// builds everything in a very specific stack:
// The square itself first
// rank and file indicator
// The piece
// The circle or square
// The legal move dot

// Each one of these components should be easily creatable and easily testable
// Once we get here, this should be easy to put all kinds of things into the square with
// simple props!

// Nextly, I do not think the square itself has ANYTHING to do with mouse events. No reason
// in any way, shape or form for the square to do anything. It must simply pass that event up
// to the caller, probably with it's square name (like "c4", or "[3,4]" or whatever we decide),
// and let the higher level components, like a board (who is the only guy with the rules engine,
// not to mention the only guy who knows if it's played, who's move it is, etc.) decide what to
// or not to do.

// I think we need "enter", "leave", "down" and "up" mouse events passed up.
// From here, exactly how to make sure a div is a square, images are sized right,
// correct colors are applied, components are rendered in a specific order on the screen,
// etc. is all yours!

// Let me know if you do not like this in any way, or if you have any questions.

// Oh, I also added eslint and prettier. Let's keep the code clean, shall we?
// Also, please make sure you write whatever relevant code doc is required for use as a
// public library. When it's internal use only, I don't care so much, but since we do intend
// for other people to use this, it would be nice to have that internal jsdoc so that future
// users of this library can get the same help we get from the libraries we use.

export default class Square extends React.Component {
  render() {
    const {
      styles,
      size,
      piece,
      pieceColor,
      showLegalMoveDot,
      squareOrCircle,
      rankAndFileIndicator
    } = this.props
    return (
      <div
        style={{
          width: size,
          height: size,
          color: 'red'
        }}
        mouseenter={() => this.onmouseenter()}
        mouseleave={() => this.onmouseleave()}
      >
        {!!rankAndFileIndicator && (
          <RankAndFileIndicator styles={styles} where={rankAndFileIndicator} />
        )}
        {!!piece && <Piece styles={styles} which={piece} color={pieceColor} />}
        {!!squareOrCircle && (
          <SquareOrCircle
            styles={styles}
            type={squareOrCircle.type}
            color={squareOrCircle.color}
            width={squareOrCircle.width}
          />
        )}
        {!!showLegalMoveDot && <LegalMoveDot styles={styles} />}
      </div>
    )
  }

  static propTypes = {
    /** The object with the various style/image/css/vector/color/etc requirements for squares, colors, pieces, legal move dots, etc. etc. etc. */
    styles: PropTypes.object.isRequired,

    /** The width of the side of the square */
    size: PropTypes.number.isRequired,
    /** A black(dark) or white(light) square */
    squareColor: PropTypes.oneOf(['b', 'w']).isRequired,

    /** The piece on the square, if there is one */
    piece: PropTypes.oneOf(['r', 'n', 'b', 'q', 'k', 'p']),
    piececolor: PropTypes.oneOf(['b', 'w']),

    /** The legal move dot, if there is one */
    showLegalMoveDot: PropTypes.bool,

    /** A square or a circle, if there is one */
    squareOrCircle: PropTypes.shape({
      type: PropTypes.oneOf(['circle', 'square']).isRequired,
      color: PropTypes.string.isRequired,
      width: PropTypes.number.isRequired
    }),

    /** Rank and file indicator if there is one */
    rankAndFileIndicator: PropTypes.oneOf([
      'tl',
      'tm',
      'tr',
      'ml',
      'mm',
      'mr',
      'bl',
      'bm',
      'br'
    ]),

    /** Fired when the mouse enters */
    onMouseIn: PropTypes.func,
    /** Fired when the mouse leaves */
    onMouseOut: PropTypes.func,
    /** Fired when the mouse button is pressed */
    onMousedown: PropTypes.func,
    /** Fired when the mouse button is released */
    onMouseUp: PropTypes.func
  }
}
