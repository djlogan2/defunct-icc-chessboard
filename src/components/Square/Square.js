import React from 'react'
import PropTypes from 'prop-types'

import Piece from '../Piece/Piece'
import Circle from '../Circle/Circle'

import styles from './square.module.css'
import LegalMove from '../LegalMove/LegalMove'

const Square = ({
  color,
  size,
  circle,
  piece,
  circleColor,
  handlePieceClick,
  pieceName,
  pieceDepends,
  legalMoves,
  updateSquareMouseDown,
  updateSquareMouseUp
}) => {
  const handleMouseUp = (event) => {
    if (typeof event === 'object' && event.button === 2) {
      updateSquareMouseUp(pieceName)
    }
  }

  const handleMouseDown = (event) => {
    if (typeof event === 'object' && event.button === 2) {
      updateSquareMouseDown(pieceName)
    }
  }

  return (
    <button
      onMouseUp={handleMouseUp}
      onMouseDown={handleMouseDown}
      onContextMenu={(e) => e.preventDefault()}
      className={styles.square}
      onClick={() => {
        handlePieceClick(pieceDepends)
      }}
      style={{
        width: size,
        height: size,
        backgroundColor: color,
        position: 'relative',
        outline: 'none'
      }}
    >
      {!!Object.keys(piece).length && (
        <Piece
          handlePieceClick={handlePieceClick}
          pieceImage={piece.image}
          description={piece.description || 'not now'}
          size={size * 0.8}
        />
      )}
      {!!circle && <Circle size={size * 0.8} strokeStyle={circleColor} />}
      {!!legalMoves && legalMoves.includes(pieceName) && (
        <LegalMove size={size * 0.3} />
      )}
    </button>
  )
}

Square.propTypes = {
  color: PropTypes.string.isRequired,
  size: PropTypes.number.isRequired,
  circle: PropTypes.bool.isRequired,
  piece: PropTypes.object
}

export default Square
