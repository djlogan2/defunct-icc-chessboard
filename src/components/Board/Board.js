import React from 'react'
import PropTypes from 'prop-types'

import Square from '../Square/Square'

import classes from './board.module.css'

const Board = ({
  size,
  boardStyle,
  pieces,
  boardSquares,
  ranksLength,
  filesLength
}) => {
  const squares = []

  for (let row = 0; row < filesLength; row++) {
    for (let col = 0; col < ranksLength; col++) {
      const color =
        row % 2
          ? col % 2
            ? boardSquares.light
            : boardSquares.dark
          : col % 2
          ? boardSquares.dark
          : boardSquares.light

      squares.push(
        <Square
          piece={pieces[row][col]}
          size={size / 8}
          key={`${row}${col}`}
          color={color}
          isLegal={false}
          circle={false}
        />
      )
    }
  }

  return (
    <div
      style={{ width: size, height: size, ...boardStyle }}
      className={classes.board}
    >
      {squares}
    </div>
  )
}

Board.propTypes = {
  size: PropTypes.number,
  pieces: PropTypes.array.isRequired,
  perspective: PropTypes.string.isRequired
}

export default Board
