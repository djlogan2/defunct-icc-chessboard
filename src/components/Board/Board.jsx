import React from 'react'
import PropTypes from 'prop-types'

import Square from '../Square/Square'
import { COLUMNS_LENGTH, ROWS_LENGTH } from '../../constants/boardConstants'
import { FIRST_COLOR, SECOND_COLOR } from '../../constants/systemConstants'

import styles from './board.module.css'

const Board = ({ size }) => {
  const squares = []

  for (let row = 0; row < ROWS_LENGTH; row++) {
    for (let col = 0; col < COLUMNS_LENGTH; col++) {
      const color =
        row % 2
          ? col % 2
            ? SECOND_COLOR
            : FIRST_COLOR
          : col % 2
          ? FIRST_COLOR
          : SECOND_COLOR

      squares.push(
        <Square size={size / 8} key={`${row}${col}`} color={color} />
      )
    }
  }

  return (
    <div style={{ width: size, height: size }} className={styles.board}>
      {squares}
    </div>
  )
}

Board.propTypes = {
  size: PropTypes.number
}

export default Board
