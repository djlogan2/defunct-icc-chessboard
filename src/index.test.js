import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import ChessBoard from './index';
import Adapter from 'enzyme-adapter-react-16';
import BoardWrapper from "./components/BoardWrapper/BoardWrapper";
import Board from "./components/Board/Board";
import Files from "./components/Lines/Files";
import Ranks from "./components/Lines/Ranks";
import Square from "./components/Square/Square";

Enzyme.configure({adapter: new Adapter()});
describe('Chessboard with props', () => {
  const container = shallow(<ChessBoard ranksSide="right" filesSide="bottom" perspective="white"/>);
  it('should render correctly', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should have no components inside', () => {
    const wrapper = mount(<ChessBoard ranksSide="right" filesSide="bottom" perspective="white"/>);

    expect(wrapper.find(<BoardWrapper/>)).toHaveLength(0);
    expect(wrapper.find(<Board/>)).toHaveLength(0);
    expect(wrapper.find(<Files/>)).toHaveLength(0);
    expect(wrapper.find(<Ranks/>)).toHaveLength(0);
  });
});

describe('BoardWrapper with props', () => {
  const container = shallow(<BoardWrapper size={100}/>);
  it('should render correctly', () => {
    expect(container.html()).toMatchSnapshot();
  });
});

describe('Board with props', () => {
  const container = shallow(<Board size={100}/>);
  it('should render correctly', () => {
    expect(container.html()).toMatchSnapshot();
  });
});

describe('Files with props', () => {
  const container = shallow(<Files width={100} side="bottom" perspective="white" />);
  it('should render correctly', () => {
    expect(container.html()).toMatchSnapshot();
  });
});

describe('Square with props', () => {
  const container = shallow(<Square color="color1" size={100} />);
  it('should render correctly', () => {
    expect(container.html()).toMatchSnapshot();
  });
});
