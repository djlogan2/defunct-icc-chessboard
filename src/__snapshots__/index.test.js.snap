// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board with props should render correctly 1`] = `"<div style=\\"width:100px;height:100px\\" class=\\"board\\"><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"even\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button><button class=\\"odd\\" style=\\"width:12.5px;height:12.5px;position:relative\\"></button></div>"`;

exports[`BoardWrapper with props should render correctly 1`] = `"<div style=\\"width:100px;height:100px\\" class=\\"boardWrapper\\"></div>"`;

exports[`Chessboard with props should render correctly 1`] = `"<div class=\\"wrapper\\"></div>"`;

exports[`Circle with props should render correctly 1`] = `"<img src=\\"https://freepngimg.com/thumb/shape/30052-8-circle-transparent-background.png\\" alt=\\"circle\\" class=\\"circle\\" style=\\"width:100px;height:auto\\"/>"`;

exports[`Files with props should render correctly 1`] = `"<div style=\\"width:100px;bottom:0;top:;position:absolute;display:flex;flex-direction:row;justify-content:space-around\\"><p>a</p><p>b</p><p>c</p><p>d</p><p>e</p><p>f</p><p>g</p><p>h</p></div>"`;

exports[`LegalMove with props should render correctly 1`] = `"<div style=\\"position:relative;left:calc(50% - 50%);height:100px;width:100px;background-color:#bbb;border-radius:50%\\"></div>"`;

exports[`Piece with props should render correctly 1`] = `"<img alt=\\"piece\\" src=\\"src\\" class=\\"piece\\" style=\\"width:100px;height:auto\\"/>"`;

exports[`Square with props should render correctly 1`] = `"<button class=\\"even\\" style=\\"width:100px;height:100px;position:relative\\"></button>"`;
